src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +52,74 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @ContentChildren(ClrLabel, { descendants: true })
+  labels: QueryList<ClrLabel>;
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    if (isPlatformBrowser(this.platformId)) {
+      this.invalidControls = Array.from(this.el.nativeElement.querySelectorAll('.ng-invalid'));
+      if (this.invalidControls.length > 0) {
+        this.invalidControls[0].focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    this.createAriaLiveElement();
+  }
+
+  private resetAriaLive() {
+    this.ariaLiveElement.innerHTML = '';
+  }
+
+  private updateAriaLive(): void {
+    if (this.invalidControls.length === 0) {
+      return;
+    }
+
+    this.ariaLiveText = this.commonStrings.keys.formErrorSummary;
+    this.resetAriaLive();
+
+    this.ariaLiveElement.textContent =
+      this.commonStrings.keys.formErrorSummary +
+      // Only use labels that have a 'for' attribute
+      this.labels
+        .filter(label => label.forAttr !== undefined)
+        // Only use labels that have a control id that matches the for attribute
+        .filter(label => this.invalidControls.filter(control => control.id === label.forAttr))
```

Comment:
> .filter(label => this.invalidControls.filter(control => control.id === label.forAttr))

Is this filter really working? Because the nested `.filter` will always return a new array. Even when there is no match found, it still returns a new array that is empty, which is still truthy.

Reply:
The array is initialized to empty so I think that should be ok. 

You are right its not working. I'll have to come up with a different way. The simple example we started with doesn't really apply any more because the @ContentChildren has all of the labels not just the ones with for attributes and the invalidIds array has objects so checking `indexFor` won't work here. 

I'll push an update soon.